File: backend/app/schemas/company.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.domain.enums import CompanySize


class CompanyCreate(BaseModel):
    name: str
    slug: str
    tax_code: str
    size: CompanySize


class CompanyRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    slug: str
    tax_code: str
    size: CompanySize
    created_at: datetime
    updated_at: datetime


class CompanyUpdate(BaseModel):
    name: str | None = None
    slug: str | None = None
    tax_code: str | None = None
    size: CompanySize | None = None

File: frontend/src/types/company.ts
export type CompanySize = 'seed' | 'startup' | 'sme' | 'enterprise'

export interface Company {
  id: string
  name: string
  slug: string
  tax_code: string
  size: CompanySize
  created_at: string
  updated_at: string
}

export interface CompanyCreateData {
  name: string
  slug: string
  tax_code: string
  size: CompanySize
}

export const COMPANY_SIZE_LABELS: Record<CompanySize, string> = {
  seed: 'Seed',
  startup: 'Startup',
  sme: 'SME',
  enterprise: 'Enterprise',
}

export const COMPANY_SIZES: CompanySize[] = [
  'seed',
  'startup',
  'sme',
  'enterprise',
]
